### PR TITLE
test: add missing tests to `utils/circular-buffer`

### LINKED ISSUE #474: Insufficient TypeScript declaration tests for `@stdlib/utils/circular-buffer`
### Description

Upon inspection of the TypeScript declaration tests for `@stdlib/utils/circular-buffer`, I noticed that the current [tests](https://github.com/stdlib-js/stdlib/blob/9f4976914e42495101d6a0717626a28a0dea2b72/lib/node_modules/%40stdlib/utils/circular-buffer/docs/types/test.ts) do not cover the entire API.

These tests should be expanded to cover the entire API surface to ensure that the current definitions are correct.

### Related Issues

None.

### Questions

No.

### Demo

_No response_

### Reproduction

```shell
N/A
```

### Expected Results

```shell
N/A
```

### Actual Results

```shell
n/A
```

### Version

_No response_

### Environments

N/A

### Browser Version

_No response_

### Node.js / npm Version

_No response_

### Platform

_No response_

### Checklist

- [X] Read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.

### PR DESCRIPTION
**Pull request by skoriop**
Toward #474.

## Description

> What is the purpose of this pull request?

This pull request adds missing tests to `utils/circular-buffer`.

## Related Issues

> Does this pull request have any related issues?

This pull request progresses #474.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by skoriop** [comment]
Hi, is there something blocking this?

**Comment by Planeshifter** [comment]
@skoriop Your changes look good and can be merged, but the linked issue includes adding tests for the remaining API surface area of the `CircularBuffer` class. Would you be able to add further tests for [its methods](https://github.com/stdlib-js/stdlib/blob/a2ed2a57008a88e74e94185443d3634b1f194223/lib/node_modules/%40stdlib/utils/circular-buffer/docs/types/index.d.ts#L49)? That would be much appreciated!

File: lib/node_modules/@stdlib/utils/circular-buffer/docs/types/test.ts [hunk 1]
```diff
@@ -26,13 +26,30 @@ import CircularBuffer = require( './index' );
 
 // TESTS //
 
-// The function returns a circular buffer instance...
+// The function returns a circular buffer instance if the constructor function is provided a first argument which is a number...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/circular-buffer/docs/types/test.ts]:
```suggestion
// The function returns a circular buffer instance...
```

Comment (kgryte) [reply lib/node_modules/@stdlib/utils/circular-buffer/docs/types/test.ts]: This change is not necessary.

File: lib/node_modules/@stdlib/utils/circular-buffer/docs/types/test.ts [hunk 2]
```diff
@@ -26,13 +26,30 @@ import CircularBuffer = require( './index' );
...
 {
 	new CircularBuffer( 3 ); // $ExpectType CircularBuffer<unknown>
 }
 
+// The function returns a circular buffer instance if the constructor function is provided a first argument which is an array-like object...
+{
+	new CircularBuffer( [ 1, 2, 3 ] ); // $ExpectType CircularBuffer<number>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/circular-buffer/docs/types/test.ts:32]: These can be placed in the block above. No need for separate tests.

**Comment by kgryte** [comment]
As is, these changes do not satisfy #474.

**Pull request merged**